scripts/generate_states.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+"""Generate library/states.c
+states.c is used to output string value of states.
```

Comment:
Reading this, I have no idea what this script does. What states are we talking about?

Reply:
moved to #4981and done， please double check.